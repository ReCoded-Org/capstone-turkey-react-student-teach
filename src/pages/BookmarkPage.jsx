import { useEffect, useState } from 'react';
import BookmarkCard from '../components/BookmarkCard';
import { data } from '../mockData';

const numberOfItemsOnClick = 4;
let arrayHoldingWholeData = [];

function BookmarkPage() {
  const [next, setNext] = useState(8);
  const [bookmarkedItemToShow, setBookmarkedItemToShow] = useState([]);

  useEffect(() => {
    loopThroughItem(0, next);
  }, []);

  console.log(bookmarkedItemToShow.length);

  // useEffect(() => {
  //   if (bookmarkedItemToShow.length < 8 && bookmarkedItemToShow.length !== 0) {
  //     handleMoreItemsOnRemoved();
  //   }
  // }, [bookmarkedItemToShow.length]);

  const loopThroughItem = (startIndex, endIndex) => {
    const slicedIndex = data.slice(startIndex, endIndex);
    arrayHoldingWholeData = [...arrayHoldingWholeData, ...slicedIndex];
    setBookmarkedItemToShow(arrayHoldingWholeData);
  };

  // const handleMoreItemsOnRemoved = () => {
  //   loopThroughItem(next, next + 1);
  //   setNext(next + 1);
  // };

  const handleMoreItems = () => {
    loopThroughItem(next, next + numberOfItemsOnClick);
    setNext(next + numberOfItemsOnClick);
  };

  const removeBookmarked = (id) => {
    const filteredBookmarks = bookmarkedItemToShow.filter((i) => i.id !== id);
    setBookmarkedItemToShow(filteredBookmarks);
    // loopThroughItem(next, next + 1);
    // setNext(next + 1);
  };

  return (
    <>
      {data && data.length ? (
        <>
          <div className="  mt-20 ml-20 ">
            <h1 className="text-[#CA7560] text-3xl font-thin ">
              Bookmarks <span className="text-4xl font-extralight">|</span>{' '}
            </h1>
            <div className="  mt-16 max-w-7xl grid grid-cols-2 gap-x-16 gap-y-14">
              {bookmarkedItemToShow.map((item) => (
                <BookmarkCard
                  key={item.id}
                  item={item}
                  removeBookmarked={removeBookmarked}
                />
              ))}
            </div>
          </div>
          <div className="text-center mt-28">
            {bookmarkedItemToShow.length < data.length && (
              <button
                onClick={handleMoreItems}
                className=" text-gray-400 -ml-28  "
              >
                load more...
              </button>
            )}
          </div>
        </>
      ) : (
        <div className="flex items-center justify-center  h-screen">
          <p className=" text-gray-400">No Bookmarked Item Found...</p>
        </div>
      )}
    </>
  );
}

export default BookmarkPage;
