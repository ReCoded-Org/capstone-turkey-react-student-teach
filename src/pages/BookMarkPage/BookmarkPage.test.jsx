import renderer from 'react-test-renderer';
import BookmarkPage from './BookmarkPage';

it('Render when text of no bookmark found removes', () => {
  const tree = renderer.create(<BookmarkPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
