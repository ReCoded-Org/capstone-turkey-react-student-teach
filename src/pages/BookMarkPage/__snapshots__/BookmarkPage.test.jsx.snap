// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render when text of no bookmark found removes 1`] = `
Array [
  <div
    className=" mt-12 mx-2  md:mt-20 md:mx-10 lg:mt-20 lg:ml-20 "
  >
    <h1
      className="text-[#CA7560] text-xl md:text-2xl lg:text-3xl font-thin "
    >
      Bookmarks
       
      <span
        className="text-2xl  md:text-3xl lg:text-4xl font-extralight"
      >
        |
      </span>
       
    </h1>
    <div
      className="  mt-16 max-w-7xl grid grid-cols-2 gap-x-5 gap-y-10 lg:gap-x-16 lg:gap-y-14"
    />
  </div>,
  <div
    className="text-center mt-28"
  >
    <button
      className=" text-gray-400 -ml-2 lg:-ml-28  "
      onClick={[Function]}
      type="button"
    >
      load more...
    </button>
  </div>,
]
`;
